rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/CompactView.java

```diff
@@ -517,14 +495,24 @@ public void onClick(MenuItemClickEvent event) {
                         }
                     }
 
-                    if (nodesource != null &&
-                        nodesource.getNodeSourceStatus().equals(NodeSourceStatus.NODES_UNDEPLOYED)) {
-                        deployItem.setEnabled(true);
+                    if (nodesource != null) {
+                        switch (nodesource.getNodeSourceStatus()) {
+                            case NODES_DEPLOYED:
+                                deployItem.setEnabled(false);
+                                undeployItem.setEnabled(true);
+                                break;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)